src/csharp/Grpc.Core/Server.cs

```diff
@@ -89,29 +94,25 @@ public void AddServiceDefinition(ServerServiceDefinition serviceDefinition)
         /// Add a non-secure port on which server should listen.
         /// Only call this before Start().
         /// </summary>
-        public int AddListeningPort(string addr)
+        /// <returns>The port on which server will be listening.</returns>
+        /// <param name="host">the host</param>
+        /// <param name="port">the port. If zero, an unused port is chosen automatically.</param>
+        public int AddListeningPort(string host, int port)
```

Comment:
I left two separate overloads to allow differentiation between secure and unsecure version of the method (because they are a very different sort of thing conceptually - and something you totally want to do right). If you like it otherwise, I can just use optional parameter.